### PR TITLE
Privmgs: iters; badges+emotes

### PR DESCRIPTION
**Pull request by Horki**
@museun please test, related to #196

### CODE REVIEW

File: src/messages/privmsg.rs [hunk 1]
```diff
@@ -46,6 +80,20 @@ impl<'a> Privmsg<'a> {
         data
     );
 
+    /// Iterator alternative to Privmsg::badges()
+    pub fn iter_badges(&self) -> BadgesIter {
+        BadgesIter {
+            items: self.tags().get("badges").unwrap().split(","),
```
Comment (museun) [inline src/messages/privmsg.rs]: These unwraps shouldn't exist. But they are easy to remove

File: src/messages/privmsg.rs [hunk 2]
```diff
@@ -46,6 +80,20 @@ impl<'a> Privmsg<'a> {
...
+        }
+    }
+
+    /// Iterator alternative to Privmsg::emotes()
+    pub fn iter_emotes(&self) -> EmotesIter {
+        EmotesIter {
+            items: self.tags().get("emotes").unwrap().split_terminator("/"),
```
Comment (museun) [inline src/messages/privmsg.rs]: These unwraps shouldn't exist. But they are easy to remove

Comment (museun) [reply src/messages/privmsg.rs]: I fixed this in 06be366

File: src/messages/privmsg.rs (hunk 1 above)
Comment (museun) [reply src/messages/privmsg.rs]: I fixed this in 06be366

**Comment by museun** [comment]
Thanks for this, I've touched it up and added unit tests. Merging it now

**Comment by Horki** [comment]
@museun great :+1: :1st_place_medal:

**Pull request merged**